hub-chart/templates/postcreatehooks/wds.yaml

```diff
@@ -0,0 +1,223 @@
+      - tenancy.kflex.kubestellar.org
+      resources:
+      - controlplanes
+      verbs:
+      - get
+      - list
+      - watch
+    - apiGroups:
+      - tenancy.kflex.kubestellar.org
+      resources:
+      - controlplanes/status
+      verbs:
+      - get
+      - patch
+      - update
+    - apiGroups:
+      - authentication.k8s.io
+      resources:
+      - tokenreviews
+      verbs:
+      - create
+    - apiGroups:
+      - authorization.k8s.io
+      resources:
+      - subjectaccessreviews
+      verbs:
+      - create
+    - apiGroups:
+      - rbac.authorization.k8s.io
+      resources:
+      - clusterroles
+      - clusterrolebindings
+      verbs:
+      - get
+      - list
+      - watch
+      - create
+      - update
+      - patch
+    - apiGroups:
+      - rbac.authorization.k8s.io
+      resources:
+      - roles
+      - rolebindings
+      verbs:
+      - get
+      - list
+      - watch
+  - apiVersion: rbac.authorization.k8s.io/v1
+    kind: ClusterRoleBinding
+    metadata:
+      name: "{{"{{.Namespace}}-{{.HookName}}"}}"
+    roleRef:
+      apiGroup: rbac.authorization.k8s.io
+      kind: ClusterRole
+      name: "{{"{{.Namespace}}-{{.HookName}}"}}"
+    subjects:
+    - kind: ServiceAccount
+      name: default
+      namespace: "{{"{{.Namespace}}"}}"
+  - apiVersion: rbac.authorization.k8s.io/v1
+    kind: Role
+    metadata:
+      name: "{{"{{.HookName}}"}}"
+    rules:
+    - apiGroups:
+      - ""
+      resources:
+      - configmaps
+      verbs:
+      - get
+      - list
+      - watch
+      - create
+      - update
+      - patch
+      - delete
+    - apiGroups:
+      - coordination.k8s.io
+      resources:
+      - leases
+      verbs:
+      - get
+      - list
+      - watch
+      - create
+      - update
+      - patch
+      - delete
+    - apiGroups:
+      - ""
+      resources:
+      - events
+      - secrets
+      - services
+      verbs:
+      - create
+      - patch
+      - update
+    - apiGroups:
+      - ""
+      resources:
+      - serviceaccounts
+      verbs:
+      - get
+      - create
+    - apiGroups:
+      - ""
+      resources:
+      - services
+      verbs:
+      - get
+    - apiGroups:
+      - apps
+      resources:
+      - deployments
+      verbs:
+      - get
+      - create
+    - apiGroups:
+      - rbac.authorization.k8s.io
+      resources:
+      - roles
+      - rolebindings
+      verbs:
+      - create
+  - apiVersion: rbac.authorization.k8s.io/v1
+    kind: RoleBinding
+    metadata:
+      name: "{{"{{.HookName}}"}}"
+    roleRef:
+      apiGroup: rbac.authorization.k8s.io
+      kind: Role
+      name: "{{"{{.HookName}}"}}"
+    subjects:
+    - kind: ServiceAccount
+      name: default
+      namespace: "{{"{{.Namespace}}"}}"
+  - apiVersion: batch/v1
+    kind: Job
+    metadata:
+      name: "{{"{{.HookName}}"}}"
+    spec:
+      template:
+        spec:
+          containers:
+          - name: "{{"{{.HookName}}-kubestellar"}}"

```

Comment:
Why include the HookName in the _container_ name? The Job is already specific to the hook.

Reply:
I think it was done by some of the hook container before, so I maintained the "style". I guess it's a preference not a mandatory requirement.